huaweicloud/services/waf/resource_huaweicloud_waf_dedicated_domain.go

```diff
@@ -238,18 +259,73 @@ func resourceWafDedicatedDomainV1Create(d *schema.ResourceData, meta interface{}
 			logp.Printf("[ERROR] error change the protection status of WAF dedicate domain %s: %s", d.Id(), err)
 		}
 	}
-	if v, ok := d.GetOk("tls"); ok {
-		updateOpts := domains.UpdatePremiumHostOpts{
-			Tls: v.(string),
+
+	if d.HasChanges("tls", "cipher", "pci_3ds", "pci_dss") {
```

Comment:
this judgment condition dose not include proxy and certificate_id, if these two parameters are changed, the method of updateWafDedicatedDomain will not be executed